### PR TITLE
[DOC] Refine Inference Doc

### PR DESCRIPTION
**Pull request by Ezra-Yu**
## Motivation

as title. not include the ImageRetrievalInferencer

## Checklist

**Before PR**:

- [ ] Pre-commit or other linting tools are used to fix the potential lint issues.
- [ ] Bug fixes are fully covered by unit tests, the case that causes the bug should be added in the unit tests.
- [ ] The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
- [ ] The documentation has been modified accordingly, like docstring or example tutorials.

**After PR**:

- [ ] If the modification has potential influence on downstream or other related projects, this PR should be tested with those projects, like MMDet or MMSeg.
- [ ] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

File: docs/en/user_guides/inference.md [hunk 1]
```diff
@@ -1,32 +1,82 @@
 # Inference with existing models
 
-MMPretrain provides pre-trained models in [Model Zoo](../modelzoo_statistics.md).
-This note will show **how to use existing models to inference on given images**.
+This note will show how to use the following APIs：
 
-As for how to test existing models on standard datasets, please see this [guide](./test.md)
+1. [**`list_models`**](mmpretrain.apis.list_models) & [**`get_model`**](<(mmpretrain.apis.get_model)>) ：list the model in MMPreTrain and get the model.
```
Comment (fangyixiao18) [inline docs/en/user_guides/inference.md:5]:
```suggestion
1. [**`list_models`**](mmpretrain.apis.list_models) & [**`get_model`**](mmpretrain.apis.get_model) ：list the model in MMPreTrain and get the model.
```

File: docs/en/user_guides/inference.md [hunk 2]
```diff
@@ -1,32 +1,82 @@
...
+# you can get model without neck, head, and output from stage 1, 2, 3 in backbone
+>>> model_headless = get_model("resnet18_8xb32_in1k", head=None, neck=None, backbone=dict(out_indices=(1, 2, 3)))
+```
+
+Then you can do the forward:
+
+```
+>>> import torch
+>>> x = torch.rand((1, 3, 224, 224))
+>>> y = model(x)
```
Comment (fangyixiao18) [inline docs/en/user_guides/inference.md:63]: better to indicate which model is used here

**Review by fangyixiao18 — CHANGES_REQUESTED** [review]
Chinese version?

File: docs/en/user_guides/inference.md (hunk 2 above)
Comment (Ezra-Yu) [reply docs/en/user_guides/inference.md:61]: done.

File: docs/en/user_guides/inference.md (hunk 1 above)
Comment (Ezra-Yu) [reply docs/en/user_guides/inference.md]: done.

File: docs/zh_CN/user_guides/inference.md [hunk 3]
```diff
@@ -1,38 +1,117 @@
...
 
-演示可以在 [demo/image_demo.py](https://github.com/open-mmlab/mmpretrain/blob/main/demo/image_demo.py) 中找到。
+```{python}
+>>> from mmpretrain import ImageClassificationInferencer
+
+>>> inferencer = ImageClassificationInferencer('resnet50_8xb32_in1k', device='cuda')
+>>> imgs = ['https://github.com/open-mmlab/mmpretrain/raw/main/demo/demo.JPEG'] * 5
+>>> results = inferencer(imgs, batch_size=2)
+>>> print(results[1]['pred_class'])
+海蛇
```
Comment (fangyixiao18) [inline docs/zh_CN/user_guides/inference.md:103]: the output of `pred_class` is Chinese?

**Review by fangyixiao18 — APPROVED** [review]

**Pull request merged**